src/runtimes/node/bundlers/nft/index.ts

```diff
@@ -140,6 +162,17 @@ const traceFilesAndTranspile = async function ({
   const normalizedDependencyPaths = [...dependencyPaths].map((path) =>
     basePath ? resolve(basePath, path) : resolve(path),
   )
+
+  if (isTypeScript) {
+    return {
+      aliases: tsAliases,
+      mainFile: getPathWithExtension(mainFile, '.js'),
+      moduleFormat: MODULE_FORMAT.ESM,
```

Comment:
We're making the v2 API ESM only, so I think hardcoding that setting here is fine.

Reply:
Although, it might still be useful for people wanting to use NFT with TypeScript on v1 functions, for some reason. I'll create an issue to do this as a follow-up.